### PR TITLE
chore: simpler tox.ini, declare dev deps in pyproject.toml

### PR DESCRIPTION
**Pull request by dimaqq**
Built on top of #1203

Sole change to review: https://github.com/juju/python-libjuju/pull/1205/commits/35499f415db3e3f8d444b8574cafb4aaca763310

### CODE REVIEW

**Comment by dimaqq** [comment]
rtd is drunk... it uses newest config yaml against older tox.ini

File: pyproject.toml [hunk 1]
```diff
@@ -34,8 +34,17 @@ dependencies = [
...
 ]
+[project.optional-dependencies]
+dev = [
+        "typing-inspect",
+        "pytest",
+        "pytest-asyncio",
+        "Twine",
+        "setuptools",  # ?
+        "pylxd",
+    ]
```
Comment (james-garner-canonical) [inline pyproject.toml]: Needs dedenting. I'll have to review the actual changes on Monday. Good luck, have fun!

Comment (dimaqq) [reply pyproject.toml]: done thx

**Comment by dimaqq** [comment]
merged into #1203

**Pull request closed without merging**